cli/loader_lib.c

```diff
@@ -161,11 +324,42 @@ __attribute__((constructor)) void jl_load_libjulia_internal(void) {
     }
 
     // Introspect to find our own path
-    const char * lib_dir = jl_get_libdir();
+    const char *lib_dir = jl_get_libdir();
 
     // Pre-load libraries that libjulia-internal needs.
     int deps_len = strlen(dep_libs);
-    char * curr_dep = &dep_libs[0];
+    char *curr_dep = &dep_libs[0];
+
+    void *cxx_handle;
+    
+#if defined(_OS_LINUX_)
+    int do_probe = 1;
+    int done_probe = 0;
+    char *probevar = getenv("JULIA_PROBE_LIBSTDCXX");
+    if (probevar) {
+        if (strcmp(probevar, "1") == 0 || strcmp(probevar, "yes"))
+            do_probe = 1;
+        else if (strcmp(probevar, "0") == 0 || strcmp(probevar, "no"))
+            do_probe = 0;
+    }
+    if (do_probe) {
+        char *cxxpath = libstdcxxprobe();
+        if (cxxpath) {
+            cxx_handle = dlopen(cxxpath, RTLD_LAZY);
+            char *dlr = dlerror();
+            if (dlr) {
+                jl_loader_print_stderr("ERROR: Unable to dlopen(cxxpath) in parent!\n");
+                jl_loader_print_stderr3("Message: ", dlr, "\n");
+                exit(1);
+            }
+            free(cxxpath);
+            done_probe = 1;
+        }
+    }
+    if (!done_probe) {
+        load_library("libstdc++.so.6", lib_dir, 1);
```

Comment:
There is a CI issue because the install path is different from the build path. Currently we call stringreplace to fix that in the `make install` step, as well as make this a static buffer in the file with a special pattern you can use for the find-replace:
```suggestion
        static const char libstdcxx_path[128]= "\0*libstdc++.so.6";
        load_library(&libstdcxx_path[2], lib_dir, 1);
```

then see how LIBJULIAINTERNAL_BUILD_DEPLIB is changed to LIBJULIAINTERNAL_INSTALL_DEPLIB in the Makefile for an example, using `\*libstdc\+\+\.so\.6$$` as the search text and `*$(call rel_path,$(libdir),$(private_libdir))/libstdc++.so.6` as the replacement text

Reply:
How about just using a define on the command line like with `GLIBCXX_LEAST_VERSION_SYMBOL`?